stl/inc/bitset

```diff
@@ -347,11 +346,11 @@ public:
     _NODISCARD _CONSTEXPR23 basic_string<_Elem, _Tr, _Alloc> to_string(
         const _Elem _Elem0 = static_cast<_Elem>('0'), const _Elem _Elem1 = static_cast<_Elem>('1')) const {
         // convert bitset to string
-        basic_string<_Elem, _Tr, _Alloc> _Str;
-        _Str.reserve(_Bits);
-
-        for (auto _Pos = _Bits; 0 < _Pos;) {
-            _Str.push_back(_Subscript(--_Pos) ? _Elem1 : _Elem0);
+        basic_string<_Elem, _Tr, _Alloc> _Str(_Bits, _Elem0);
+        for (size_t _Pos = 0; _Pos < _Bits; ++_Pos) {
+            if (_Subscript(_Bits - 1 - _Pos)) {
+                _Str[_Pos] = _Elem1;
+            }

```

Comment:
Does this become branchy after this change?

If it becomes branchy, I think `_Str[_Pos] += _Subscript(_Bits - 1 - _Pos)` or `_Str[_Pos] += _Subscript(_Bits - 1 - _Pos) ? _Elem1 : _Elem0` can keep it branchless.

(Assuming `string::resize_and_overwrite` is not applied, otherwise it is _Str[_Pos] += _Subscript(_Bits - 1 - _Pos) ? _Elem1 : _Elem0`)

Reply:
The codegen does appear to be branchy. (Interestingly, MSVC appears willing to fully unroll a `bitset<100>`'s loop, which I did not expect.) I think we should defer such changes to a followup PR to investigate their impact in isolation, as the current changes are a strict improvement.

Reply:
Also filed #3858 for followup.